gensim/models/atmodel.py

```diff
@@ -0,0 +1,1017 @@
+            (self.num_terms, self.num_topics, self.num_authors, self.decay, self.chunksize)
+
+    def init_empty_corpus(self):
+        """
+        Initialize an empty corpus. If the corpora are to be treated as lists, simply
+        initialize an empty list. If serialization is used, initialize an empty corpus
+        of the class `gensim.corpora.MmCorpus`.
+
+        """
+        if self.serialized:
+            # Tnitialize the corpus as a serialized empty list.
+            # This corpus will be extended in self.update.
+            MmCorpus.serialize(self.serialization_path, [])  # Serialize empty corpus.
+            self.corpus = MmCorpus(self.serialization_path)  # Store serialized corpus object in self.corpus.
+        else:
+            # All input corpora are assumed to just be lists.
+            self.corpus = []
+
+    def extend_corpus(self, corpus):
+        """
+        Add new documents in `corpus` to `self.corpus`. If serialization is used,
+        then the entire corpus (`self.corpus`) is re-serialized and the new documents
+        are added in the process. If serialization is not used, the corpus, as a list
+        of documents, is simply extended.
+        
+        """
+        if self.serialized:
+            # Re-serialize the entire corpus while appending the new documents.
+            if isinstance(corpus, MmCorpus):
+                # Check that we are not attempting to overwrite the serialized corpus.
+                assert self.corpus.input != corpus.input, 'Input corpus cannot have the same file path as the model corpus (serialization_path).'
+            corpus_chain = chain(self.corpus, corpus)  # A generator with the old and new documents.
+            copyfile(self.serialization_path, self.serialization_path + '.tmp')  # Make a temporary copy of the file where the corpus is serialized.
+            self.corpus.input = self.serialization_path + '.tmp'  # Point the old corpus at this temporary file.
+            MmCorpus.serialize(self.serialization_path, corpus_chain)  # Re-serialize the old corpus, and extend it with the new corpus.
+            self.corpus = MmCorpus(self.serialization_path)  # Store the new serialized corpus object in self.corpus.
+            remove(self.serialization_path + '.tmp')  # Remove the temporary file again.
+        else:
+            # self.corpus and corpus are just lists, just extend the list.
+            # First check that corpus is actually a list.
+            assert isinstance(corpus, list), "If serialized == False, all input corpora must be lists."
+            self.corpus.extend(corpus)
+
+
+    def compute_phinorm(self, ids, authors_d, expElogthetad, expElogbetad):
+        """Efficiently computes the normalizing factor in phi."""
+        phinorm = np.zeros(len(ids))
+        expElogtheta_sum = np.zeros(self.num_topics)
+        for a in xrange(len(authors_d)):
+            expElogtheta_sum += expElogthetad[a, :]
```

Comment:
No, because it is not a sum over all authors. `authors_d` is a list of the authors in the current document (document `d`).